packages/txservice/src/transaction.ts

```diff
@@ -260,14 +260,19 @@ export class Transaction {
       this.receipt = result.value;
     }
 
+    // Sanity checks.
     if (this.receipt == null) {
       // Receipt is undefined or null. This normally should never occur.
       throw new TransactionServiceFailure("Unable to obtain receipt: ethers responded with null.", {
         method,
         receipt: this.receipt,
       });
     } else if (this.receipt.status === 0) {
+      // This should never occur. We should always get a TransactionReverted error in this event.
       throw new TransactionServiceFailure("Transaction was reverted but TransactionReverted error was not thrown.");
```

Comment:
We should include the hash on the error here

Reply:
We should include error in context in general, good catch